src/Tinker.jl

```diff
@@ -49,12 +66,28 @@ function init_gui(image::AbstractArray; name="Tinker")
         end
     end
 
-    # rectangle draw function
-    function drawrect(ctx, rect, color)
-        set_source(ctx, color)
-        rectangle(ctx, rect[1], rect[2], rect[3], rect[4])
-        stroke(ctx)
-    end;
+    # holds x zoom level
+    xzoom = map(zr) do r
+        100*r.fullview.x.right/(r.currentview.x.right-(r.currentview.x.left-1))
+    end
+
+    # hold y zoom level
+    yzoom = map(zr) do r
+        100*r.fullview.y.right/(r.currentview.y.right-(r.currentview.y.left-1))
+    end
+
+    # performs proportional zoom in (x2 each time)
+    function zoomin()
+        push!(zr, zoom(value(zr), 0.5))
+        println("zoom x%: ", value(xzoom),"\n zoom y%: ", value(yzoom))
+        # why doesn't this update after push?
+    end
+
+    # performs proportional zoom out (x0.5 each time)
+    function zoomout() # doesn't perfectly undo zoomin: problem
```

Comment:
Maybe add a `@show value(zr)` or something somewhere. Note that by default a `ZoomRegion` gets created using [RoundingIntegers](https://github.com/JuliaMath/RoundingIntegers.jl) (see the `RInt` in https://github.com/JuliaGizmos/GtkReactive.jl/blob/f160ac0d8c0c655fc4bee7a1f6a9b0681f228cd4/src/graphics_interaction.jl#L333-L340) to make sure the zoom region are always integers, so things are definitely rounded. You could, of course, create a `ZoomRegion{Float64}` if you prefer, but then you might have to round elsewhere.